crates/bevy_math/src/bounding/bounded2d.rs

```diff
@@ -0,0 +1,284 @@
+    };
+    let b = Aabb2d {
+        min: Vec2::new(-2., -0.5),
+        max: Vec2::new(0.75, 1.),
+    };
+    let merged = a.merged(&b);
+    assert!((merged.min - Vec2::new(-2., -1.)).length() < 0.0001);
+    assert!((merged.max - Vec2::new(1., 1.)).length() < 0.0001);
+}
+
+#[test]
+fn test_aabb2d_padded() {
+    let a = Aabb2d {
+        min: Vec2::new(-1., -1.),
+        max: Vec2::new(1., 1.),
+    };
+    let padded = a.padded((Vec2::ONE, Vec2::Y));
+    assert!((padded.min - Vec2::new(-2., -2.)).length() < 0.0001);
+    assert!((padded.max - Vec2::new(1., 2.)).length() < 0.0001);
+}
+
+#[test]
+fn test_aabb2d_shrunk() {
+    let a = Aabb2d {
+        min: Vec2::new(-1., -1.),
+        max: Vec2::new(1., 1.),
+    };
+    let shrunk = a.shrunk((Vec2::ONE, Vec2::Y));
+    assert!((shrunk.min - Vec2::new(-0., -0.)).length() < 0.0001);
+    assert!((shrunk.max - Vec2::new(1., 0.)).length() < 0.0001);
+}
+
+/// A bounding circle
+pub struct BoundingCircle {
+    /// The center of the bounding circle
+    pub center: Vec2,
+    /// The radius of the bounding circle
+    pub radius: f32,
+}
+
+impl BoundingVolume for BoundingCircle {
+    type Position = Vec2;
+    type Padding = f32;
+
+    #[inline(always)]
+    fn center(&self) -> Self::Position {
+        self.center
+    }
+
+    #[inline(always)]
+    fn visible_area(&self) -> f32 {
+        std::f32::consts::PI * self.radius * self.radius
+    }
+
+    #[inline(always)]
+    fn contains(&self, other: &Self) -> bool {
+        if other.center == self.center {
+            other.radius <= self.radius
+        } else {
+            let furthest_point = (other.center - self.center).length() + other.radius;
+            furthest_point <= self.radius
+        }
+    }
+
+    #[inline(always)]
+    fn merged(&self, other: &Self) -> Self {
+        if other.center == self.center {
+            Self {
+                center: self.center,
+                radius: self.radius.max(other.radius),
+            }
+        } else {
+            let diff = other.center - self.center;
+            let length = diff.length();
+            let dir = diff / length;
+
+            Self {
+                center: self.center + dir * ((length + other.radius - self.radius) / 2.),
+                radius: (length + self.radius + other.radius) / 2.,
+            }
+        }
+    }
+
+    #[inline(always)]
+    fn padded(&self, amount: Self::Padding) -> Self {
+        Self {
+            center: self.center,
+            radius: self.radius + amount,
+        }
+    }
+
+    #[inline(always)]
+    fn shrunk(&self, amount: Self::Padding) -> Self {
+        debug_assert!(self.radius >= amount);
+        Self {
+            center: self.center,
+            radius: self.radius - amount,
+        }
+    }
+}
+
+#[test]
+fn test_bounding_circle_area() {
+    let circle = BoundingCircle {
+        center: Vec2::ONE,
+        radius: 5.,
+    };
+    assert!((circle.visible_area() - 78.5398).abs() < 0.001);
```

Comment:
We should pull out 0.0001 into an EPSILON constant.

Reply:
This, and all other tests, should probably be in a private test module, gated behind the cfg(test) flag to mildly improve compile times.